### PR TITLE
add Stackable Callbacks to dbBindCallback

### PR DESCRIPTION
**Pull request by jokoho48**
- Add Stackable Callbacks to dbBindCallback
- Call Callback Directly when Result is ready
- Update git Submodules

PS: please be tame this is the first time in 2 years writing any C++ 😅

### CODE REVIEW

File: addons/main/config.cpp [hunk 1]
```diff
@@ -1,16 +1,16 @@
 class CfgPatches {
-	class intercept_template_plugin { //Change this
```
Comment (dedmen) [inline addons/main/config.cpp]: oops, I really didn't put much work into this i guess :D

File: src/logger.cpp [hunk 2]
```diff
@@ -28,7 +28,7 @@ void Logger::logThread(intercept::types::r_string message) {
 }
 
 void Logger::pushTimestamp(std::ostream& str) const {
-      std::time_t t = std::time(0);   // get time now
+    std::time_t t = std::time(0);   // get time now
     std::tm* now = std::localtime(&t);
```
Comment (dedmen) [inline src/logger.cpp]: actually, while copy pasting these a few days ago, both of these are actually unused, but doesn't matter

File: src/main.cpp [hunk 3]
```diff
@@ -58,12 +58,18 @@ void intercept::on_frame() {
     for (auto& it : Threading::get().completedAsyncTasks) {
         __itt_task_begin(domainMain, __itt_null, __itt_null, main_on_frame_callback);
 
-        if (!it->data->callback.is_nil() && it->data->res) {
+        if (it->data->callbacks.size() != 0 && it->data->res) {
```
Comment (dedmen) [inline src/main.cpp]:
```suggestion
        if (!it->data->callbacks.empty() && it->data->res) {
```
Atleast I think empty() is a method

File: src/main.cpp [hunk 4]
```diff
@@ -58,12 +58,18 @@ void intercept::on_frame() {
...
             logMessageWithTime("task callback");
             auto gd_res = new GameDataDBResult();
             gd_res->res = it->data->res;
-
-            sqf::call(it->data->callback, { gd_res, it->data->callbackArgs });
+            for (size_t i = 0; i < it->data->callbacks.size(); i++)
+            {
```
Comment (dedmen) [inline src/main.cpp]: wrong brace layout, needs to be on same line.
also why not store the arg together with the callback, and use a for each loop?

File: src/res.h [hunk 5]
```diff
@@ -114,8 +114,8 @@ class GameDataDBAsyncResult : public game_data {
         std::future<bool> fut;
         mariadb::result_set_ref res;
         r_string statementName; //Is null if this was not a statement from config
-        game_value callback;
-        game_value callbackArgs;
+        std::vector<game_value> callbacks;
```
Comment (dedmen) [inline src/res.h]: use 
auto_array<std::pair<game_value, game_value>>
for callbacks and args in single array.

File: src/logger.cpp (hunk 2 above)
Comment (jokoho48) [reply src/logger.cpp]: I removed it then

File: src/main.cpp (hunk 3 above)
Comment (jokoho48) [reply src/main.cpp]: done

File: src/main.cpp (hunk 4 above)
Comment (jokoho48) [reply src/main.cpp]: fixed

File: src/res.h (hunk 5 above)
Comment (jokoho48) [reply src/res.h]: changed

**Comment by dedmen** [comment]
Don't understand why I didn't get this idea before, this is great.
I'll check the rest tomorrow and fix it myself if problems, I hope I remember tomorrow ^^

**Comment by jokoho48** [comment]
i can set you a reminder in the ACE slack 🤣

File: src/main.cpp [hunk 6]
```diff
@@ -58,12 +58,15 @@ void intercept::on_frame() {
...
         __itt_task_begin(domainMain, __itt_null, __itt_null, main_on_frame_callback);
 
-        if (!it->data->callback.is_nil() && it->data->res) {
+        if (!it->data->callbacks.is_empty() && it->data->res) {
             logMessageWithTime("task callback");
             auto gd_res = new GameDataDBResult();
             gd_res->res = it->data->res;
-
-            sqf::call(it->data->callback, { gd_res, it->data->callbackArgs });
+            for each (std::pair<game_value, game_value> callback in it->data->callbacks) {
```
Comment (dedmen) [inline src/main.cpp]: for (auto& [code, arg] : it->data->callbacks)
sqf::call(code, {gd_res, arg});


atleast I hope structured binding works properly with auto_array

Comment (jokoho48) [reply src/main.cpp]: done that too

**Comment by dedmen** [comment]
shrink_to_fit is unnecessary because
- clear() already clears memory
- The whole result including callbacks will be deleted shortly after the callbacks were processed

**Comment by dedmen** [comment]
> fatal: could not read Username for 'https://github.com': No such device or address

No idea whats's wrong with appveyor..

**Pull request merged**